lib/SILOptimizer/Differentiation/LinearMapInfo.cpp

```diff
@@ -117,6 +117,7 @@ LinearMapInfo::createBranchingTraceDecl(SILBasicBlock *originalBB,
     branchingTraceDecl->setAccess(AccessLevel::Internal);
   }
   file.addTopLevelDecl(branchingTraceDecl);
+  file.getParentModule()->clearLookupCache();
```

Comment:
Out of curiosity: Why is this needed?

Reply:
We need to clear cache after something is added to the top-level, otherwise `ModuleDecl::lookupValue()` won't pick up the added decl.
Previously, this was called in `getOrCreateSynthesizedFile()`, but it was technically a wrong place because the user calls `ModuleDecl::lookupValue()` before adding a top-level decl in it, it'd break.

Reply:
Thanks for the explanation. If you want, you can add a comment with it for future reference.